### ISSUE TITLE
note that conversion doesn't guarantee a deepcopy

### ISSUE THREAD

**Issue by DirectXMan12** [kubernetes-sigs/controller-runtime#546]
The conversion interface currently doesn't say anything about deepcopy vs no deepcopy.  We should take a position on this.

cc @droot 

/kind documentation

**Comment by mengqiy** [comment]
The `Convertible` interface requires it to be a `runtime.Object` which implies it implements `DeepCopyObject() Object`. Did I miss anything?

```
type Convertible interface {
	runtime.Object
	ConvertTo(dst Hub) error
	ConvertFrom(src Hub) error
}
```

**Comment by droot** [comment]
@DirectXMan12 can you elaborate more on deepcopy vs no deepcopy ?

**Comment by DirectXMan12** [comment]
Right, but we don't specify if `ConvertTo` and `ConvertFrom` *need* to call `DeepCopy` themselves.  Can `ConvertTo` and `ConvertFrom` safely re-use pointers to data from the source object?  Is it expected that anyone who calls deepcopy doesn't care what becomes of source (this is true for conversion webhooks, but we should add a note).

TL;DR: in Rust terminology, does `ConvertTo` take an immutable reference to self, or does it consume self?

**Comment by vincepri** [comment]
/help

Need to mostly add documentation.
